src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,45 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  box-sizing: border-box;
+  width: $widthCalendar;
+  &__day {
+    width: $widthDay;
+    height: $heightDay;
+    margin: $marginBetweenDays / 2;
+    box-sizing: border-box;
+    background-color: $backgroundDay;
+    border: 1px solid #000;
+    position: relative;
+    transition: all 0.5s ease 0s;
+    &:hover {
+      cursor: pointer;
+      background-color: #ffbfcb;
+      transform: translateY(-20px);
+    }
+
+    @for $day from 1 through 31 {
+      &:nth-child(#{$day})::before {
+        content: '#{$day}';
+        display: inline-block;
+        font-size: $fontSize;
+        position: absolute;
+        left: 50%;
+        top: 50%;
+        transform: translate(-50%, -50%);
+      }
+    }
+  }
+
+  @each $dayWeek, $i in $daysOfWeek {
+    &--start-day-#{$dayWeek} .calendar__day:first-child {
+      margin-left: $widthDay * $i + $marginBetweenDays * ($i + 0.5);
```

Comment:
The calculation for margin-left in the &--start-day-#{$dayWeek} selector might need verification. Ensure that the formula $widthDay * $i + $marginBetweenDays * ($i + 0.5) correctly aligns the first day of the week according to your design specifications.